stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.xml

```diff
@@ -89,10 +89,56 @@
         >
             <attribute name="t-if">move_line.id</attribute>
         </xpath>
-        <!-- Remove "All items could not be shipped..." sentence -->
+        <!-- Remove "All items could not be shipped..." sentence
+            Substitute it by a message indicating if all goods have been delivered. -->
         <xpath expr='//div[hasclass("page")]/p[last()]' position="attributes">
             <attribute name="t-if">False</attribute>
         </xpath>
+        <xpath expr='//div[hasclass("page")]/p[last()]' position="after">
+            <t t-if="o.picking_type_id.code == 'outgoing'">
+                <p>Remaining to deliver:</p>
+                <t t-if="o.demand_is_completed()">
+                    <p>All the ordered items were delivered.</p>
+                </t>
+                <t t-else="">
+                    <table class="table table-sm" name="remaining_to_deliver_table">
+                        <tbody>
+                            <t
+                                t-set="pending_moves"
+                                t-value="get_remaining_to_deliver_data(o)"
+                            />
+                            <tr t-foreach="pending_moves" t-as="pending_move">
+                                <td>
+                                    <t
+                                        t-if="pending_move['is_header']"
+                                    ><![CDATA[<strong>]]></t>
+                                    <span t-esc="pending_move['concept']" />
+                                    <t
+                                        t-if="pending_move['is_header']"
+                                    ><![CDATA[</strong>]]></t>
+                                </td>
+                                <td style="text-align: right">
+                                    <t
+                                        t-if="pending_move['is_header']"
+                                    ><![CDATA[&nbsp;]]></t>
```

Comment:
?

Reply:
This is because of ```black```, that doesn't recognise ```&nbsp;``` as a valid entity, and fails to validate.

Reply:
then you have to quote it: `&amp;nbsp;`

Reply:
What's the problem with using CDATA? I don't get it.

Reply:
it makes it unreadable and you don't need it.

Reply:
also, I wonder if you need the space at all...

Reply:
To me, <![CDATA[&nbsp;]]> is more readable than &amp;nbsp;

Reply:
Nevermind, I see the space is not needed. I remove it.